app/models/spree/volume_price.rb

```diff
@@ -14,6 +14,16 @@ class Spree::VolumePrice < ActiveRecord::Base
 
   validate :range_format
 
+  def self.for_variant(variant, user: nil)
+    role_id = user ? user.resolve_role : nil
```

Comment:
I kinda think it actually provides better readability by using a local var vs. putting an expression into a method call.